orttraining/orttraining/core/graph/training_op_defs.cc

```diff
@@ -2634,6 +2634,41 @@ Example 4:
             return ONNX_NAMESPACE::FunctionBodyHelper::BuildFunctionProto(functionProto, schema, body, {onnx_opset_13});
           });
 
+  ONNX_CONTRIB_OPERATOR_SCHEMA(QuickGeluGrad)
+      .SetDomain(kMSDomain)
+      .SinceVersion(1)
+      .SetSupportLevel(OpSchema::SupportType::EXPERIMENTAL)
+      .SetDoc("QuickGeluGrad")
+      .Attr("alpha", "Alpha value.", AttributeProto::FLOAT, 1.702f)
+      .AllowUncheckedAttributes()
+      .Input(0, "dY", "The gradient tensor from output.", "T")
+      .Input(1, "X", "The input tensor. ", "T")
+      .Output(0, "dX", "Gradient of the input.", "T")
+      .TypeConstraint("T", {"tensor(float16)", "tensor(float)", "tensor(double)", "tensor(bfloat16)"},
+                      "Constrain input and output types to float tensors.")
+      .TypeAndShapeInferenceFunction(ONNX_NAMESPACE::propagateShapeAndTypeFromFirstInput)
+      .SetContextDependentFunctionBodyBuilder(
+          [](const FunctionBodyBuildContext& ctx, const OpSchema& schema, FunctionProto& functionProto) {
+            auto* tp = ctx.getInputType(0);
+            if ((tp == nullptr) || (!tp->has_tensor_type())) return false;
+            auto elem_type = (ONNX_NAMESPACE::TensorProto_DataType)tp->tensor_type().elem_type();
+            auto* alpha_attr = ctx.getAttribute("alpha");
+            float alpha = (alpha_attr != nullptr) ? alpha_attr->f() : 1.702f;
+            std::vector<FunctionBodyHelper::NodeDef> body{ONNX_NAMESPACE::Const("Alpha", alpha, elem_type),
+                                                          ONNX_NAMESPACE::Const("One", 1.f, elem_type),
+                                                          {{"CX"}, "Mul", {"Alpha", "X"}},
+                                                          {{"SigmoidCX"}, "Sigmoid", {"CX"}},
+                                                          {{"OneMinusSigmoidCX"}, "Sub", {"One", "SigmoidCX"}},
+                                                          {{"Mul1"}, "Mul", {"CX", "SigmoidCX"}},
+                                                          {{"Mul2"}, "Mul", {"Mul1", "OneMinusSigmoidCX"}},
+                                                          {{"Add1"}, "Add", {"Mul2", "SigmoidCX"}},
+                                                          {{"dX"}, "Mul", {"dY", "Add1"}}};
+            OperatorSetIdProto onnx_opset_13;
```

Comment:
opset 15? since training already supports it?

Reply:
Already removed the function body.